api/v1alpha2/types.go

```diff
@@ -406,11 +406,25 @@ type AvailabilityZone struct {
 	Enabled *bool   `json:"enabled,omitempty"`
 }
 
+// Image defines information about the image to use for VM creation.
+// There are three ways to specify an image: by ID, by publisher, or by Shared Image Gallery.
+// If specifying an image by ID, only the ID field needs to be set.
+// If specifying an image by publisher, the Publisher, Offer, SKU, and Version fields must be set.
+// If specifying an image from a Shared Image Gallery, the SubscriptionID, ResourceGroup,
+// Gallery, Name, and Version fields must be set.
 type Image struct {
-	Publisher string `json:"publisher"`
-	Offer     string `json:"offer"`
-	SKU       string `json:"sku"`
-	Version   string `json:"version"`
+	Publisher *string `json:"publisher,omitempty"`
```

Comment:
are these fields optional? If not, the type should be string vs *string.

Reply:
@juan-lee -- They're optional-ish. Only required when specifying a published image.

Here's what I wrote for the `Image` field, which I'm going to write a doc for in short order:
```go
// Image defines information about the image to use for VM creation.
// There are three ways to specify an image: by ID, by publisher, or by Shared Image Gallery.
// If specifying an image by ID, only the ID field needs to be set.
// If specifying an image by publisher, the Publisher, Offer, SKU, and Version fields must be set.
// If specifying an image from a Shared Image Gallery, the SubscriptionID, ResourceGroup,
// Gallery, Name, and Version fields must be set.
```

Reply:
Just thinking out loud on how to make this more user friendly. This is where ObjectReference or claim style approach is nice. Each of those permutations could be expressed as a different type. Another option would be for image to have a type and use RuntimeExtension or a map.

Reply:
@justaugustus I filed issue #293 to track this improvement in the future.